### ISSUE TITLE
[CAL-915] embed times in mail

### ISSUE THREAD

**Issue by FUF1** [calcom/cal.com#6533]
### Is your proposal related to a problem?

No, it is a feature request.

### Describe the solution you'd like

Although cal.com is in many ways superior to calendly they have one feature I really find useful. You can select dates and times for an event and copy it directly to insert to a traditional email or you can embed it in websites.
![calendly_2](https://user-images.githubusercontent.com/69962736/212942276-5b10cfe8-549b-4b2b-9d0f-f30266064d0d.png)
![calendly_1](https://user-images.githubusercontent.com/69962736/212942278-d677dd2a-ad81-4f7c-b02b-d7f688d6ce9f.png)

### Describe alternatives you've considered

I could go via calendly. But I prefer cal.com due to its opensource philosophy and the generally better features.

### Additional context

The implementation could be better and there should as well be an option that people can select a date and provide a different time proposal or provide a different date & time proposal altogether. Some more advanced options and customizations (also regarding design) than calendly provides would be helpful. Glad to think with the developers about conceptual solutions to this problem. _Please, keep up the phenomenal work and the good philosophy!_

<sub>[CAL-915](https://linear.app/calcom/issue/CAL-915/embed-times-in-mail)</sub>

**Comment by PeerRich** [comment]
yea this is planned

**Comment by FUF1** [comment]
Amazing, thanks. As an additional comment/idea: For me it would have been helpful many times in the past to have an option to set a specific calendar (from google calendar for example) and make people book only dates where the timeslot lies within the events from that calendar. This way it is easy to give some quick options without making a preselection or a special availability preset with date overrides (or at least in general it would be useful to be able to define a specific daterange in which a date can be booked).

**Comment by davidmucci** [comment]
@PeerRich Also, check out this... https://www.mailmodo.com/use-case/embed-calendly-inside-email/

Embedding the whole flow from date selection to confirmation right inside the email would be really cool!

**Comment by FUF1** [comment]
Just found this by accident: [savvycal.com](https://savvycal.com). Seems to have many of the above mentioned features. (Actually wasn't sure in the first moment if it is a spin-off of cal.com due to the similar webdesign.)

**Comment by PeerRich** [comment]
yep, its on the roadmap now! help wanted if anyone is down

**Comment by FUF1** [comment]
> yep, its on the roadmap now! help wanted if anyone is down

I am very willing to help, but full disclaimer that I am not a properly trained programmer. If there is anything you think can be of beneficial help, I am willing to learn and help out.

**Comment by PeerRich** [comment]
@FUF1 there are many non-technical parts you can help with, such as translations, helping others self-host, testing things in PRs, going through issues and finding duplicates, testing the product and opening issues if you found bugs, sharing the repo with friends, post on social media, ... ❤️

**Comment by FUF1** [comment]
Very glad to hear that. I already started promoting cal.com to my company and others when I started using it. I guess following the issue section in github will present such opportunities.

**Comment by ShaneMaglangit** [comment]
Hey! I would love to give this one a try.

I'm thinking of including this as another embed option. The layout would generally be the same as the rest. However, the fields would be tailored to fit with this specific embedding type.

It might be a bit tricky to get the whole booking flow embedded in the email because we're building our pages with NextJS and can only embed static HTML inside emails. Thus, I'm thinking of keeping the first iteration simple by just having static content with links to cal.com booking pages (exactly like Calendly). @PeerRich would love to hear your thoughts.

> We would have an option for email embed here
![image](https://user-images.githubusercontent.com/53674742/225130065-e882317d-98d8-43fc-9667-22599278b808.png)

> Fields on the left  will be tailored for selecting dates/times. On the right, we would only have an html and preview tabs.
![image](https://user-images.githubusercontent.com/53674742/225131140-88f868d6-5702-486b-b709-1450d3bc58c5.png)

**Comment by ShaneMaglangit** [comment]
Submitted a bid for this, but I'm really bad an estimating value for these stuffs as I primarily contribute as an hobby 😆. Let me know if there's a more reasonable amount that I can bid.

**Comment by ShaneMaglangit** [comment]
The alternative of course, would be to just have a clone of the booking page inside this modal. The only change would be to allow multi-select on the time ranges.

![image](https://user-images.githubusercontent.com/53674742/225134887-12e3bd83-6dcc-4249-8abe-6f4e2e25f845.png)

**Comment by PeerRich** [comment]
@ShaneMaglangit i am bad at estimating too. i could update the bid, maybe someone made a lower bit. how much time would you think to dedicate?

**Comment by ShaneMaglangit** [comment]
> @ShaneMaglangit i am bad at estimating too. i could update the bid, maybe someone made a lower bit. how much time would you think to dedicate?

Hey, not at all,. I think that the bid amount is great 👌. I just personally don't think I have the time to commit on this one.

**Comment by ShaneMaglangit** [comment]
> @ShaneMaglangit i am bad at estimating too. i could update the bid, maybe someone made a lower bit. how much time would you think to dedicate?

Hello again @PeerRich. The reason why I initially thought that this would take a lot of involvement and time was due to how my initial proposal required me to rebuild/compress the whole AvailabilityPage logic into the modal.

Got a bit of free time this weekend and wanted to explore a different approach. LMKWYT about this approach:

- Email Embed selection is still added on top of the 3 existing options.
- Upon selection, it will redirect to the booking page with an `exportType=email` params.
  - This has the following effect:
    - Add some added header contents for the page (title and export/copy button).
    - SlotPicker becomes multiselect.
    
I tried the other approach of embeding this availability page in a modal but it produce some UI challenges and some logic disconnect since its within an iframe.

![image](https://user-images.githubusercontent.com/53674742/226112199-f6c59297-3f7c-450d-92fe-fe3fba26f50a.png)
